### ISSUE TITLE
`#onInventoryMoveItemEvent()` taking excessively long when tile entities are ticked

### ISSUE THREAD

**Issue by natereprogle** [TerrorByteTW/Honeypot#187]
**Describe the bug**
A member of my Discord server reported an issue with the `onInventoryMoveItemEvent()` method, which is in charge of preventing hoppers and droppers from pulling and/or pushing into Honeypot inventories. Tile entity ticks take quite a long time on the server when Honeypot's storage method is set to SQLite (No longer the default, but was a very recent change so many are still using it). In the provided screenshot, `tickBlockEntities()` is roughly 31.73% of the each tick's processing. Honeypot's `InventoryMoveItemEventListener` takes 22.36% of that, meaning that Honeypot is responsible for **~70%** of the processing time that `tickBlockEntities()` takes.

**To Reproduce**
1. Set Honeypot to SQLite (It defaults to PDC in newer versions, but SQLite was the default for a long time)
2. Create a 💩ton of Hoppers
3. Load them suckers
4. Open the spark profiler and see how long Honeypot is taking to check each Hopper if it's a Honeypot each time they tick.

**Expected behavior**
The server should have only slightly higher-than-normal tick times due to Hoppers inherently being laggy.

**Screenshots**
![image](https://github.com/TerrorByteTW/Honeypot/assets/4064388/f7c1a2b7-1e73-41f1-88e6-b21477f9063e)

**Version Info (please complete the following information):**
 - Honeypot Version: [e.g. 1.5]: 3.3.2
 - Java Version [e.g. JDK 1.18]: Unknown
 - MC Server Version [e.g. 1.18.2]: 1.20.6-2232-f1c87e4
 - MC Server Type [e.g. Paper, Spigot, etc.]: Purpur

**Additional context**

**Comment by natereprogle** [comment]
Currently working on this, the plan for now is to ignore Hoppers entirely until I can figure out a more efficient way to handle this.

**Comment by natereprogle** [comment]
The solution that I've come up with is to use native block locking for containers. In vanilla, block locking disallows players from accessing an inventory but still allows hoppers and droppers to interact with them. However, because we can detect this in code and it's a vanilla feature which uses NBT data, we can utilize this for the `#onInventoryMoveItemEvent()` listener without lag issues. 

Currently, there is a bug with this feature though that causes players items to be completely deleted if they attempt to place an item into a Honeypot container that is locked. This is due to the inventory of the container not being updated when the player closes it, and I cannot for the life of me figure out why this is happening. So, WIP still.

TL;DR -- The lag issue is fixed, but it introduced a bug which can delete player items. No bueno

**Comment by natereprogle** [comment]
I ended up undoing this and going with Spatial Indexing instead using SQLite. There were some bugs with this method that not even the Paper devs themselves could assist me with. Essentially, blockstate wasn't getting updated and there was literally no reason why.

In any case, Spatial Indexing seems to have drastically reduced tick times for Honeypot, so I think this method was a success! I'm asking the reporter to test now.

![image](https://github.com/user-attachments/assets/95918604-5d57-450b-b592-f8eafb9b95ce)
![image](https://github.com/user-attachments/assets/6e26feac-4c51-42c4-9aa4-aac076696690)
